### PR TITLE
Bug fix with change delivery option in order

### LINKED ISSUE #23389: Bug when set other delivery option in order via admin panel
#### Describe the bug

I started changing the delivery in the order through the admin panel and noticed that the delivery price was calculated incorrectly

#### Steps to Reproduce

1. Make carrier with range 0 to 100 price 50, range 100 to 100500 price 0
2. Make product with combination, price product 20, impact of price + 200 for example
3. Add product in cart, to make order
4. Go to admin panel in maked order, change delivery option on other method and now need back to old delivery option.
5. Now you can to look, what price stay 50 instead 0

#### Additional information

* PrestaShop version: 1.7.6.5
* PHP version: 7.2

### PR DESCRIPTION
**Pull request by fest73**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Bug when changing delivery option in order via admin panel
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/23389
| How to test?      | Here example for playback https://github.com/PrestaShop/PrestaShop/issues/23389
| Possible impacts? | no

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23390)

### CODE REVIEW

**Review by matthieu-rolland — COMMENTED** [review]
Thank you for your contribution @dariusakafest

Seems like php cs fixer is not happy, could you run this command from the root of your prestashop project ? and then commit the modification:

`./vendor/friendsofphp/php-cs-fixer/php-cs-fixer fix classes/order/Order.php`

**Comment by fest73** [comment]
@matthieu-rolland its ok now)

File: classes/order/Order.php [hunk 1]
```diff
@@ -2747,7 +2747,7 @@ public function refreshShippingCost()
             $new_cart->updateQty(
                 $product['product_quantity'],
                 (int) $product['product_id'],
-                null,
+                (int) $product['product_attribute_id'],
```
Comment (Progi1984) [inline classes/order/Order.php]:
```suggestion
                $product['product_attribute_id'] ? (int) $product['product_attribute_id'] : null,
```

Comment (sowbiba) [reply classes/order/Order.php]:
```suggestion
                isset($product['product_attribute_id']) ? (int) $product['product_attribute_id'] : null,
```

**Review by PierreRambaud — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Comment by Robin-Fischer-PS** [comment]
Hi @dariusakafest ,

As @hibatallahAouadni said in your issue, we cannot reproduce your bug on 1.7.7.2 (I managed to reproduce the bug 1.7.6.5 but not on 1.7.7.x or develop, so it seems it's already fixed).

Can you try to reproduce your bug and share us a screenrecord of this on the 1.7.7.x or develop branch ? Otherwise we will not be able to test and validate your PR, since the issue cannot be reproduced.

Thanks !

**Comment by matks** [comment]
Hello, it seems this PR is now stale and we get no answer from author.

I close this PR, following our [review guidelines](https://devdocs.prestashop.com/1.7/project/maintainers-guide/reviewing-pull-requests/) "Pull Requests may be closed after 30 days of inactivity". If however you are still interested in it, please ping us :) we can reopen. 😉

**Pull request closed without merging**